server/src/main/java/org/elasticsearch/common/network/NetworkModule.java

```diff
@@ -122,6 +125,7 @@ public NetworkModule(
         NamedXContentRegistry xContentRegistry,
         NetworkService networkService,
         HttpServerTransport.Dispatcher dispatcher,
+        BiConsumer<BasicHttpRequest, ThreadContext> dispatcherContext,
```

Comment:
We can choose to merge the `dispatcherContext` in the `HttpServerTransport.Dispatcher` interface, if we opt to go with presented approach where the `RestController` furnishes the thread context supplier, but it's the http server transport that employs it, because the Security plugin can alter the latter but not the former.